libs/configuration-variables-resolver/src/VariablesLoader/ConfigurationVariablesLoader.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\ConfigurationVariablesResolver\VariablesLoader;
+
+use Keboola\ConfigurationVariablesResolver\ComponentsClientHelper;
+use Keboola\ConfigurationVariablesResolver\Exception\UserException;
+use Psr\Log\LoggerInterface;
+
+class ConfigurationVariablesLoader
+{
+    public function __construct(
+        private readonly ComponentsClientHelper $componentsHelper,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    /**
+     * @param array{variables_id?: string|null, variables_values_id?: string|null} $configuration
+     * @param non-empty-string|null $variableValuesId
+     * @param array{values?: list<array{name: scalar, value: scalar}>|null}|null $variableValuesData
+     * @return array<non-empty-string, string>
+     */
+    public function loadVariables(array $configuration, ?string $variableValuesId, ?array $variableValuesData): array
+    {
+        if ($variableValuesId && !empty($variableValuesData['values'])) {
+            throw new UserException('Only one of variableValuesId and variableValuesData can be entered.');
+        }
+
+        $variablesId = $configuration['variables_id'] ?? null;
+        $variablesValuesId = $configuration['variables_values_id'] ?? null;
+
+        if (!$variablesId) {
+            return [];
+        }
+
+        $variablesConfiguration = $this->componentsHelper->getVariablesConfiguration($variablesId);
+        $variablesData = $this->loadVariablesData(
+            $variablesId,
+            $variablesValuesId,
+            $variableValuesId,
+            $variableValuesData,
+        );
+
+        $values = [];
+        foreach ($variablesData['values'] as $row) {
+            $key = (string) $row['name'];
+            $value = (string) $row['value'];
```

Comment:
Tohle pretypovani `value` na `string` se driv delalo az v context classe (https://github.com/keboola/platform-libraries/blob/main/libs/configuration-variables-resolver/src/VariablesContext.php#L32). Jelikoz Vault vraci vzdy vsechno jako `string` a v config variables (jestli jsme spravne pochopil) taky nikdy nebyla implementovana podpora pro jiny typ nez `string`, prislo mi rozumny to natypovani presunout uz sem.